src/core.js

```diff
@@ -425,6 +425,16 @@ jQuery.extend({
 	support: support
 });
 
+// jsHint would error on this code due to the Symbol not being defined in ES5.
+// Defining this global in .jshintrc would create a danger of using the global
+// unguarded in another place, it seems safer to just disable jsHint for these
+// three lines.
+/* jshint ignore: start */
+if ( typeof Symbol === "function" && Symbol.iterator ) {
```

Comment:
Current situation is as follows:

1. Latest stable Firefox, Chrome & Opera support `Symbol`, `Symbol.iterator` & `for-of` on any collections.
2. IE11 support neither `Symbol` nor `for-of` but Edge supports both.
3. Safari 8 doesn't support `Symbol`; it supports `for-of` only on Arrays. WebKit Nightly supports `Symbol`, `Symbol.iterator` & `for-of` on any collections.

I don't think we're in a lot of danger that any browser will arrive with that much broken `Symbol` global judging by the above info.